src/pool/pool_disjoint.c

```diff
@@ -25,6 +27,65 @@
 #include "utils_log.h"
 #include "utils_math.h"
 
+/* Disjoint pool CTL implementation */
+struct ctl disjoint_ctl_root;
+static UTIL_ONCE_FLAG ctl_initialized = UTIL_ONCE_FLAG_INIT;
+
+static int CTL_READ_HANDLER(name)(void *ctx, umf_ctl_query_source_t source,
+                                  void *arg, size_t size,
+                                  umf_ctl_index_utlist_t *indexes,
+                                  const char *extra_name,
+                                  umf_ctl_query_type_t queryType) {
+    (void)source, (void)indexes, (void)queryType, (void)size, (void)extra_name;
+    disjoint_pool_t *pool = (disjoint_pool_t *)ctx;
+
+    if (arg == NULL) {
+        return -1;
+    }
+
+    strncpy((char *)arg, pool->params.name, size);
+    return 0;
+}
+
+static int CTL_WRITE_HANDLER(name)(void *ctx, umf_ctl_query_source_t source,
+                                   void *arg, size_t size,
+                                   umf_ctl_index_utlist_t *indexes,
+                                   const char *extra_name,
+                                   umf_ctl_query_type_t queryType) {
+    (void)source, (void)indexes, (void)queryType, (void)size, (void)extra_name;
+    disjoint_pool_t *pool = (disjoint_pool_t *)ctx;
+    if (arg == NULL) {
+        return -1;
+    }
+
+    strncpy(pool->params.name, (char *)arg, sizeof(pool->params.name) - 1);
+    return 0;
+}
+
+static const umf_ctl_node_t CTL_NODE(disjoint)[] = {CTL_LEAF_RW_no_arg(name),
+                                                    CTL_NODE_END};
+
+static void initialize_disjoint_ctl(void) {
+    CTL_REGISTER_MODULE(&disjoint_ctl_root, disjoint);
+}
+
+umf_result_t disjoint_pool_ctl(void *hPool, int operationType, const char *name,
+                               void *arg, size_t size,
+                               umf_ctl_query_type_t queryType) {
+    (void)operationType, (void)queryType;
+    utils_init_once(&ctl_initialized, initialize_disjoint_ctl);
+
+    const char *prefix = "disjoint.";
+    const char *name_wo_prefix = strstr(name, "disjoint.");
+
+    // Check if the name has the prefix
+    if ((name_wo_prefix = strstr(name, prefix)) == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
```

Comment:
not sure what this code does, but it's incorrect. there should never be "disjoint" in string here.  And there should be no special string parsing outside of ctl

Reply:
Done